### DISCUSSION TITLE
Make public key available at .well-known/jwks.json

### DISCUSSION THREAD

**Discussion by janoschp** [ory/kratos#3866] [Q&A] [answered]
After following this tutorial https://www.ory.sh/docs/identities/session-to-jwt-cors#end-to-end-example and successfully creating a private+public key set and patching the identity-config, I am able to generate Json web tokens from a given session. I now try to validate this token on my server and would like to retrieve the public key from https://{project.slug}.projects.oryapis.com/.well-known/jwks.json . But unfortunately my public key is not available from this URL, isn't it supposed to be available there? Is there any way to make it available? I would like to avoid hardcoding the public key into my backend.

**Comment by nikitabarskov** [comment, marked as answer]
> After following this tutorial https://www.ory.sh/docs/identities/session-to-jwt-cors#end-to-end-example and successfully creating a private+public key set and patching the identity-config, I am able to generate Json web tokens from a given session. I now try to validate this token on my server and would like to retrieve the public key from https://{project.slug}.projects.oryapis.com/.well-known/jwks.json . But unfortunately my public key is not available from this URL, isn't it supposed to be available there? Is there any way to make it available? I would like to avoid hardcoding the public key into my backend.

Hey, @janoschp

Not sure if this helps, but I see that Ory docs says it's possible to do so now:

https://www.ory.sh/docs/hydra/jwks#the-role-of-well-knownjwksjson

Let me know if you have figure out the way to serve it, thank you!
